### PR TITLE
Dropdown and refactoring

### PR DESCRIPTION
**Pull request by OlafSzmidt**
**Things implemented and fixed:**

- [x] Dropdown implemented will all levels existing in Assets/Scenes/Levels directory.
- [x] Generation of new levels can now be done via a button
- [x] Scenes are unloaded, new scenes are correctly set up with lighting and camera settings.
- [x] The bug where changes were not being saved after adding obstacles is now fixed.

**This should fix the following fully:** 

- Level builder - 'Create level' button  #35 
- Level builder - LightSubmenu instead of conventional LightSubMenu #43  

**And the following partially:** 

- Level Builder - Generating obstacles not seen as a modification by unity #52 

**And raises the following tasks to do next:**

- Level Builder - Remove scenes from editor after closing #54  
- Level Builder - Save button #53 
- Level Builder - Refactor LevelBuilderWindow #48

### CODE REVIEW

File: UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs [hunk 1]
```diff
@@ -21,9 +22,11 @@
  * 
  */ 
 
+[System.Serializable]
```
Comment (danalex97) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs]: Is this needed?

File: UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs [hunk 2]
```diff
@@ -74,15 +77,79 @@ public void OnGUI()
...
+			lightSettings.color = new Color (1.0f, 0.956f, 0.839f, 1.0f);
+			lightSettings.intensity = 1.0f;
+			lightSettings.shadowBias = 0.05f;
+			lightSettings.shadowNormalBias = 0.4f;
+			lightSettings.shadowNearPlane = 0.2f;
+			directionalLight.transform.position = new Vector3 (0.0f, 3.0f, 0.0f);
+			directionalLight.transform.rotation = Quaternion.Euler(50.0f, -30.0f, 0.0f);
+
+			if (!EditorSceneManager.SaveScene (newScene, "Assets/Scenes/Levels/Level" + currentNoOfLevels + ".unity")) {
+				Debug.Log ("The savescene line failed!!!!!!!!!!!!!");
```
Comment (danalex97) [inline UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs]: NIT: Maybe we should use [LogWarning](https://docs.unity3d.com/ScriptReference/Debug.LogWarning.html) or LogError?

Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs]: Yes. That is definitely the plan. I didn't want to make this PR too large (for obvious reasons :)))) that's why I left some unfinished things that are suggested (or will be) in #41.

Comment (danalex97) [reply UnityProject/Assets/Editor/LevelBuilder/GUI/LevelBuilderWindow.cs]: Fair :+1:

File: UnityProject/Assets/Editor/LevelBuilder/Util/SceneHandler.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	{
+		Scene createdScene = EditorSceneManager.NewScene (NewSceneSetup.EmptyScene, NewSceneMode.Additive);
+
+		return createdScene;
+	} // createMethod
+
+	// Returns the number of Scenes that are currently present in the Assets/Scenes/Levels directory.
+	public static int countScenes()
+	{
+		levelsFetched = AssetFetcher.GetLevels ();
```
[the remark this answers is not in the document]
Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/Util/SceneHandler.cs]: @ramonfmir MonoDevelop does this automatically. I think it's a coding convention for C-sharp. I hate it :((((((((((

File: UnityProject/Assets/Editor/LevelBuilder/Util/SceneHandler.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	private static IList<string> levelsFetched;
+	private static string[] levelsArray;
+
+	// Calling this returns an empty Scene object that can be further manipulated and saved.
+	public static Scene createScene()
+	{
+		Scene createdScene = EditorSceneManager.NewScene (NewSceneSetup.EmptyScene, NewSceneMode.Additive);
+
+		return createdScene;
+	} // createMethod
```
[the remark this answers is not in the document]
Comment (OlafSzmidt) [reply UnityProject/Assets/Editor/LevelBuilder/Util/SceneHandler.cs]: We don't. Good point. It's a habit of mine from Java. Will be fixed with #59.

**Pull request merged**